component/mcsd/component.go

```diff
@@ -473,3 +485,169 @@ func extractResourceIDFromURL(entry fhir.BundleEntry) string {
 
 	return ""
 }
+
+// buildOrganizationKey creates a composite key from organization ID and VersionId for deduplication.
+// The key includes both the ID and VersionId (if available) to differentiate between versions.
+func buildOrganizationKey(org *fhir.Organization) string {
+	if org == nil || org.Id == nil {
+		return ""
+	}
+	key := *org.Id
+	if org.Meta != nil && org.Meta.VersionId != nil {
+		key = key + ":" + *org.Meta.VersionId
+	}
+	return key
+}
+
+// findOrganizationKeyByID builds a map of organization keys by their ID to help with partOf reference lookups.
+// This is needed because partOf references typically only contain the ID, not the VersionId.
+func findOrganizationKeyByID(orgMap map[string]*fhir.Organization, orgID string) string {
+	for key, org := range orgMap {
+		if org.Id != nil && *org.Id == orgID {
+			return key
+		}
+	}
+	return ""
+}
+
+// findParentOrganizationWithURA finds a parent organization with a URA identifier and all organizations linked to it.
+// It loops through ALL entries to find organizations with URA identifiers, then selects the one with the most
```

Comment:
Does it really take the amount of child relationships into account? It think it just builds the index independent of the amount of relationships, right?